sdk/servicebus/service-bus/test/batchReceiver.spec.ts

```diff
@@ -1191,3 +1192,57 @@ describe("Batching - disconnects", function(): void {
     refreshConnectionCalled.should.be.greaterThan(0, "refreshConnection was not called.");
   });
 });
+
+describe("Batching - close() respects an in progress init()", () => {
+  afterEach(() => {
+    return afterEachTest();
+  });
+
+  it("close() while init() is happening", async () => {
+    sbClient = getServiceBusClient();
+
+    const { receiverClient, senderClient } = await getSenderReceiverClients(
+      sbClient,
+      TestClientType.UnpartitionedQueue,
+      TestClientType.UnpartitionedQueue
+    );
+
+    // force the link open - doesn't affect our tests below but it does
+    // let me mock out the proper method that I need to prove things are working
+    await senderClient.createSender().open();
+
+    const receiver = receiverClient.createReceiver(ReceiveMode.receiveAndDelete);
+
+    const origReceiverFn = sbClient["_context"].connection["createReceiver"];
+
+    let closePromise: Promise<void> | undefined;
+    let createdReceiver: RheaReceiver | undefined;
+
+    sbClient["_context"].connection["createReceiver"] = async (...options: any) => {
+      // close() will be blocked at this point
+      closePromise = receiver.close();
+
+      const result = await Promise.race([
+        closePromise,
+        delay(2000, "delay_should_win_because_close_is_blocked")
```

Comment:
So, the init invoked from `receiveMessages` holds the lock, and thus `close` is blocked?

Reply:
Yes, let me rework it a bit to make that more clear.

Reply:
I added in a comment - does that make things stand out more?

Reply:
Sounds good.